### PR TITLE
ScalafmtRunner: remove dialect parsing hack

### PR DESCRIPTION
**Pull request by kitbellew**

### CODE REVIEW

**Comment by tgodzik** [comment]
I think we need to first release new Scalameta? I would wait until we get more info about possible bugs and @dos65 PR is merged.

**Comment by kitbellew** [comment]
> I think we need to first release new Scalameta? I would wait until we get more info about possible bugs and @dos65 PR is merged.

of course! this is for visibility, so we don't forget it, after upgrading scalameta.

**Comment by tgodzik** [comment]
@kitbellew are you looking into the scalameta upgrade? Or should I ? I've seen some tests start to fail for 4..4.16.

**Comment by kitbellew** [comment]
> @kitbellew are you looking into the scalameta upgrade? Or should I ? I've seen some tests start to fail for 4.4.16.

i have it ready, some of the failing tests led to scalameta/scalameta#2334 and scalameta/scalameta#2335. i tagged 4.4.17, should send the pr soon.

File: scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala [hunk 1]
```diff
@@ -87,6 +81,7 @@ object ScalafmtRunner {
     val scala213 = Scala213
       .withAllowTrailingCommas(true)
     val default = scala213
+      .withAllowTraitParameters(true) // not true but backwards compatible
```
Comment (tgodzik) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala:84]: Why do we need it? Aren't trait params only available from Scala 3 ?

Comment (kitbellew) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala:84]: we have two options:
- keep this
- remove this test: https://github.com/scalameta/scalafmt/blob/master/scalafmt-tests/src/test/resources/unit/Dialect.stat#L20
thoughts?

Comment (tgodzik) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala:84]: I would remove the tests, although I am not sure why it was needed in the first place.

Comment (kitbellew) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala:84]: will do.

**Review by tgodzik — APPROVED** [review]

**Pull request merged**